File: back/src/test/java/com/tdevilleduc/urthehero/back/utils/TestUtils.java
package com.tdevilleduc.urthehero.back.utils;

import com.tdevilleduc.urthehero.back.model.Story;
import org.testcontainers.shaded.org.apache.commons.lang.RandomStringUtils;

import java.util.Random;

public class TestUtils {

    private static Random random = new Random();

    public static Story createRandomStory() {
        Integer storyId = random.nextInt();
        String storyTitle = RandomStringUtils.random(20);
        Integer storyAuthorId = random.nextInt();
        Integer storyFirstPageId = random.nextInt();
        String storyDetailedText = RandomStringUtils.random(20);
        String storyImage = RandomStringUtils.random(20);
        return new Story(storyId, storyTitle, storyAuthorId, storyFirstPageId,storyDetailedText, storyImage);
    }

    public static Story createStory(Integer authorId, Integer firstPageId) {
        Integer storyId = random.nextInt();
        String storyTitle = RandomStringUtils.random(20);
        String storyDetailedText = RandomStringUtils.random(20);
        String storyImage = RandomStringUtils.random(20);
        return new Story(storyId, storyTitle, authorId, firstPageId,storyDetailedText, storyImage);
    }
}

File: back/src/test/java/com/tdevilleduc/urthehero/back/utils/TestUtils.kt
package com.tdevilleduc.urthehero.back.utils

import com.tdevilleduc.urthehero.back.model.PageDTO
import com.tdevilleduc.urthehero.back.model.PersonDTO
import com.tdevilleduc.urthehero.back.model.Story
import com.tdevilleduc.urthehero.back.model.StoryDTO
import org.testcontainers.shaded.org.apache.commons.lang.RandomStringUtils
import java.util.*

object TestUtils {
    private val random: Random = Random()
    fun createRandomStory(): Story {
        val id = random.nextInt()
        val storyAuthorId = random.nextInt()
        val storyFirstPageId = random.nextInt()
        val title = RandomStringUtils.random(20)
        val detailedText = RandomStringUtils.random(20)
        val image = RandomStringUtils.random(20)
        return Story(id, storyAuthorId, storyFirstPageId, title, detailedText, image)
    }

    fun createStory(authorId: Int, firstPageId: Int): StoryDTO {
        val id = random.nextInt()
        val title = RandomStringUtils.random(20)
        val detailedText = RandomStringUtils.random(20)
        val image = RandomStringUtils.random(20)
        return StoryDTO(id, title, authorId, firstPageId, detailedText, image)
    }

    fun createPerson(): PersonDTO {
        val id = random.nextInt()
        val login = RandomStringUtils.random(20)
        val displayName = RandomStringUtils.random(20)
        val email = RandomStringUtils.random(20)
        val password = RandomStringUtils.random(20)
        return PersonDTO(id, login, displayName, email, password)
    }

    fun createPage(): PageDTO {
        val id = random.nextInt()
        val text = RandomStringUtils.random(20)
        val image = RandomStringUtils.random(20)
        return PageDTO(id, text, image)
    }
}